lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -0,0 +1,635 @@
+    log_mode=dict(type='str', choices=['stderr', 'file', 'syslog'], default='syslog'),
+    filter_logger=dict(type='bool', default=True),
+    debug=dict(type='bool', default=False),
+)
+
+AZURE_CREDENTIAL_ENV_MAPPING = dict(
+    profile='AZURE_PROFILE',
+    subscription_id='AZURE_SUBSCRIPTION_ID',
+    client_id='AZURE_CLIENT_ID',
+    secret='AZURE_SECRET',
+    tenant='AZURE_TENANT',
+    ad_user='AZURE_AD_USER',
+    password='AZURE_PASSWORD'
+)
+
+AZURE_TAG_ARGS = dict(
+    tags=dict(type='dict'),
+    purge_tags=dict(type='bool', default=False),
+)
+
+AZURE_COMMON_REQUIRED_IF = [
+    ('log_mode', 'file', ['log_path'])
+]
+
+CIDR_PATTERN = re.compile("(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1"
+                          "[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))")
+
+AZURE_SUCCESS_STATE = "Succeeded"
+AZURE_FAILED_STATE = "Failed"
+
+AZURE_MIN_VERSION = "2016-03-30"
+
+HAS_AZURE = True
+HAS_AZURE_EXC = None
+
+try:
+    from enum import Enum
+    from msrest.serialization import Serializer
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute import __version__ as azure_compute_version
+    from azure.mgmt.network.models import PublicIPAddress, NetworkSecurityGroup, SecurityRule, NetworkInterface, \
+        NetworkInterfaceIPConfiguration, Subnet
+    from azure.common.credentials import ServicePrincipalCredentials, UserPassCredentials
+    from azure.mgmt.network.network_management_client import NetworkManagementClient,\
+                                                             NetworkManagementClientConfiguration
+    from azure.mgmt.resource.resources.resource_management_client import ResourceManagementClient,\
+                                                                         ResourceManagementClientConfiguration
+    from azure.mgmt.storage.storage_management_client import StorageManagementClient,\
+                                                             StorageManagementClientConfiguration
+    from azure.mgmt.compute.compute_management_client import ComputeManagementClient,\
+                                                             ComputeManagementClientConfiguration
+    from azure.storage.cloudstorageaccount import CloudStorageAccount
+except ImportError, exc:
+    HAS_AZURE_EXC = exc
+    HAS_AZURE = False
+
+
+def azure_id_to_dict(id):
+    pieces = re.sub(r'^\/', '', id).split('/')
+    result = {}
+    index = 0
+    while index < len(pieces) - 1:
+        result[pieces[index]] = pieces[index + 1]
+        index += 1
+    return result
+
+
+class AzureRMModuleBase(object):
+
+    def __init__(self, derived_arg_spec, bypass_checks=False, no_log=False,
+                 check_invalid_arguments=True, mutually_exclusive=None, required_together=None,
+                 required_one_of=None, add_file_common_args=False, supports_check_mode=False,
+                 required_if=None, supports_tags=True):
+

```

Comment:
bcoca and I came to an agreement that **kwargs would be well used here to blackbox the arguments that are forwarded onto AnsibleModule.  So this signature should look like:
``` python
def __init__(self, derived_arg_spec, supports_tags=True, required_if=None, **kwargs):
```